### PR TITLE
Publish artifacts directly to sonatype.

### PR DESCRIPTION
**Pull request by anuraaga**
This is the same configuration we use for instrumentation so is just bringing it over. While we haven't had anywhere near as many stability problems in this repo, it's good to be consistent, and fixes an issue we're seeing related to jars on artifactory. A lot less config, a lot more kotlin DSL to achieve a similar result :)

Ran `./gradlew snapshot` locally and it seems to check out (including javadoc / sources jar which currently artifactory doesn't have for whatever reason).
https://oss.sonatype.org/content/repositories/snapshots/io/opentelemetry/opentelemetry-api/0.15.0-SNAPSHOT/

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -99,7 +99,7 @@ We strongly recommend using our published BOM to keep all dependency versions in
   <repositories>
     <repository>
       <id>oss.sonatype.org-snapshot</id>
```
Comment (anuraaga) [inline README.md:101]: I didn't have to change this :P

**Review by iNikem — APPROVED** [review]

**Review by jkwatson — APPROVED** [review]
let's give it a whirl. I'll revert if it doesn't work when merged.

**Comment by jkwatson** [comment]
unfortunately, we lost `bom` publishing with this change

**Comment by jkwatson** [comment]
I think we actually had lost bom publishing with an earlier change.

**Pull request merged**